### PR TITLE
Make FN editable

### LINKED ISSUE #89: Display "name" or "nickname" instead of "first name" + "last name"
I have some contacts that have the same first and last names and differ only in their middle name. When I attempt to select one from `mutt`'s menu, or the native `khard` menu, the displayed names seem to be derived from the "first name" + "last name" instead of the actual name.

For example, I have the following contact. 

```
Name: John Foo Smith
First name : John
Additional : Foo
Last name  : Smith
```

and also

```
Name: John Bar Smith
First name : John
Additional : Bar
Last name  : Smith
```

From khard's native interface, it's difficult to differentiate between the two.

```
$ khard modify smith
Select contact for Modify action
Address book: owncloud
Index    Name          Phone                 E-Mail                   UID   
1        John Smith    CELL: 1234 567 890    WORK: jbsmith@baz.com    49    
2        John Smith    CELL: 1234 567 891    WORK: jfsmith@baz.com    5a4   
Enter Index:
```

Preferably, the basic contact "name" should be displayed. Another good option would be to display the "nickname" instead, although I guess this would need to be an option since not everyone would like that. Personally, the latter seems useful to me, but the first option is probably "safer".

### LINKED ISSUE #171: Add FN (Full Name) of vCard to display configuration
Also mentioned in #89 by @madduck , there's a "FN" field in vCard.

Not only same first name and last name people exists in real world, there are also some countries put last name first and first name afterwards, i.e. China, Japan, etc.

As a result, it would be better if 'FN' is configurable in khard. Thx.

### PR DESCRIPTION
**Pull request by lucc**
See #171 for the initial discussion.

closes #171 
closes #89

### CODE REVIEW

File: khard/helpers.py [hunk 1]
```diff
@@ -202,6 +202,10 @@ def get_new_contact_template(supported_private_objects=None):
 
     # create template
     return dedent("""
+        # Every contact must contain a formatted name, it will be autofilled
+        # from the full name below if not given.
+        Formatted name :
```
Comment (hydrargyrum) [inline khard/helpers.py]: Isn't it rather `full name` when it's the input, and `formatted name` when it's generated by first name/last name templating?

Comment (lucc) [reply khard/helpers.py]: It represents the value in the `FN` field of the vcard.  And that is
> formatted text corresponding to the name

https://tools.ietf.org/html/rfc6350#section-6.2.1

File: khard/khard.py [hunk 2]
```diff
@@ -474,18 +479,14 @@ def get_contacts(address_books, query, method="all", reverse=False,
...
-    :param method: the search method, one of "all", "name" or "uid"
-    :type method: str
-    :param reverse: reverse the order of the returned contacts
-    :type reverse: bool
-    :param group: group results by address book
-    :type group: bool
-    :param sort: the field to use for sorting, one of "first_name", "last_name"
-    :type sort: str
+    :param  address_books: the address books to search
+    :type address_books:
```
Comment (hydrargyrum) [inline khard/khard.py]: why did you remove the type here?

File: khard/carddav_object.py [hunk 3]
```diff
@@ -1171,6 +1167,10 @@ def _process_user_input(self, input):
             contact_data.get("Prefix", ""), contact_data.get("First name", ""),
             contact_data.get("Additional", ""),
             contact_data.get("Last name", ""), contact_data.get("Suffix", ""))
+        if "Formatted name" in contact_data:
+            self.formatted_name = contact_data.get("Formatted name")
+        if not self.formatted_name:
+            self.formatted_name = ""
```
Comment (hydrargyrum) [inline khard/carddav_object.py]: From reading the diff, I know this is supposed to trigger the `property.setter`, but if someone later on reads this code, they will probably miss the context and think this `if not x: x = ""` is useless, or worse, remove it. Maybe this could deserve a little code comment?

File: khard/khard.py (hunk 2 above)
Comment (lucc) [reply khard/khard.py]: Editing mistake, fixed

File: khard/carddav_object.py (hunk 3 above)
Comment (lucc) [reply khard/carddav_object.py]: good idea

**Comment by hydrargyrum** [comment]
@lucc  For some reasons, it seems the tests are failing.

**Comment by d7415** [comment]
> @lucc For some reason, it seems the tests are failing.

That failure seems to have been fixed in 28366a9c3b553015c0145baaab6d8f23b140edc4. A rebase (or the eventual merge) should fix it.
